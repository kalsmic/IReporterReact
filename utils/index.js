export const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const baseURL = 'https://ireporterv3.herokuapp.com/api/v3';

export const isAuthenticated = () => {
  const token = sessionStorage.getItem('iReporterToken');
  if (token) { return true; }
  return false;
};
