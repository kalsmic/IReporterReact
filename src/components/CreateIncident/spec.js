import { shallow } from 'enzyme';
import React from 'react';
import { CreateIncident, mapDispatchToProps, mapStateToProps } from './index';
import { findByTestAttr } from '../../../utils';

describe('CreateIncident Component', () => {
  const newIncidentFn = jest.fn();
  const props = {
    newIncident: newIncidentFn,
    error: {},
    message: '',
    incident: {},
    isLoading: false,
  };
  const wrapper = shallow(<CreateIncident {...props} />);

  it('should render without crushing', () => {
    expect(wrapper)
      .toMatchSnapshot();
  });

  it('should handle the onchange event', () => {
    const event = {
      target: {
        name: 'title',
        value: 'my first title',
      },
    };

    wrapper.instance()
      .handleChange(event);
    expect(wrapper.instance().state.title)
      .toBe(event.target.value);
  });

  it('should handle the onSubmit event', () => {
    const instance = wrapper.instance();
    wrapper.setState({
      title: '',
      comment: ''
    });
    const event = {
      target: {
        type: 'submit',
        name: 'login',
      },
      preventDefault: jest.fn(),
    };

    instance.handleSubmit(event);
    expect(instance.props.newIncident)
      .not
      .toBeCalled();
    wrapper.setState({
      title: 'Incident Title',
      comment: 'My comment',

      incidentLocation: [12, 90],
      latlng: {
        lat: 51.505,
        lng: -0.09,
      },
      hasLocation: false,
      incidentType: 'red-flag',
    });
    instance.handleSubmit(event);
    expect(instance.props.newIncident)
      .toBeCalled();
    expect(instance.props.newIncident)
      .toBeCalledWith(
        'Incident Title', 'My comment', [12, 90], 'red-flag',
      );
  });

  it('should show messages to the user', () => {
    const error = {
      title: 'titleError',
      comment: 'commentError',
      location: 'locationError',
    };
    wrapper.setProps({
      error,
      message: 'incident created Successfully',
      isLoading: true
    });
    expect(findByTestAttr(wrapper, 'titleError')
      .text())
      .toBe(error.title);
    expect(findByTestAttr(wrapper, 'commentError')
      .text())
      .toBe(error.comment);
    expect(findByTestAttr(wrapper, 'locationError')
      .text())
      .toBe(error.location);
    expect(findByTestAttr(wrapper, 'locationError')
      .text())
      .toBe(error.location);
    expect(findByTestAttr(wrapper, 'locationError')
      .text())
      .toBe(error.location);
  });


  it('should map state to props', () => {
    const mockedState = {
      incidentReducer: {
        message: 'Created Successfully',
        isLoading: false,
        error: '',
        incident: {},
      },
    };
    const state = mapStateToProps(mockedState);
    expect(state.message)
      .toBe('Created Successfully');
    expect(state.isLoading)
      .toBe(false);
    expect(state.error)
      .toBe('');
  });

  it('should map dispatch to props', () => {
    const mockedDispatch = jest.fn();

    mapDispatchToProps(mockedDispatch)
      .newIncident();
    expect(mockedDispatch)
      .toHaveBeenCalled();
  });
});
