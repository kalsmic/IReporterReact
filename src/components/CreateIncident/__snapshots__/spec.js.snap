// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateIncident Component should render without crushing 1`] = `
<Root>
  <h3
    className="text-blue bd_line_bottom"
  >
    New Incident Record
  </h3>
  <form
    className="incident__form"
    onSubmit={[Function]}
  >
    <span
      className="incident__form__label"
    >
      Type of Incident : 
    </span>
    <select
      className="incident__form__input"
      name="incidentType"
      onChange={[Function]}
      required={true}
      value=""
    >
      <option
        value=""
      >
        Please Select Incident type
      </option>
      <option
        value="red-flag"
      >
        Red-flag
      </option>
      <option
        value="intervention"
      >
        Intervention
      </option>
    </select>
    <span
      className="incident__form__label"
    >
      Title
    </span>
    <input
      className="incident__form__input"
      name="title"
      onChange={[Function]}
      placeholder="What the incident is about ?"
      required={true}
      type="text"
      value=""
    />
    <span
      className="incident__form__label"
    >
      Comment 
    </span>
    <Quill
      className="incident__form__input"
      id="commentWys"
      modules={Object {}}
      onChange={[Function]}
      theme="snow"
      value=""
    />
    <IncidentMap
      handleMapClick={[Function]}
      hasLocation={false}
      latlng={
        Object {
          "lat": 0.3428767,
          "lng": 32.5915189,
        }
      }
      mapClassName="incident__form__map"
      mapReference={
        Object {
          "current": null,
        }
      }
    />
    <div
      className="incident__form__footer"
    >
      <div
        className="incident__form__footer__message"
      />
      <button
        className="btn bg-green text-white "
        type="submit"
      >
        Add Incident Record
         
      </button>
    </div>
  </form>
</Root>
`;
